open_academy/models/session.py

```diff
@@ -1,13 +1,48 @@
-from odoo import models, fields
+from odoo import models, fields, api
+from odoo.exceptions import ValidationError
 
 
 class Session(models.Model):
     _name = 'session'
     _description = 'Session'
     course_id = fields.Many2one('course', required=True)
-    instructor_id = fields.Many2one('res.partner')
+    instructor_id = fields.Many2one('res.partner', domain=("['|',"
+                                                                "('is_instructor', '=', 'True'),"
+                                                                "('category_id.name', 'like', 'Teacher'),"
+                                                            "]"))
     attendee_ids = fields.Many2many('res.partner')
     name = fields.Char(required=True)
     start_date = fields.Date()
     duration = fields.Float()
-    number_seat = fields.Integer()
+    number_seat = fields.Integer(default=1, required=True)
+    taken_seat = fields.Integer(default=0)
```

Comment:
Remove this field, use `attendee_ids` to compute the percentage instead, because you can have 10 taken seat, but not attendees, and viceversa